build-scripts/substitute-local-appservices.gradle

```diff
@@ -0,0 +1,32 @@
+logger.lifecycle("[local-appservices] adjusting project to use locally published application-services modules")
+
+// Inject mavenLocal repository. This is where we're expected to publish modules.
+repositories {
+    mavenLocal()
+}
+
+configurations.all { config ->
+    if (config.isCanBeResolved()) {
+        config.resolutionStrategy { strategy ->
+            dependencySubstitution {
+                all { dependency ->
+                    if (!(dependency.requested instanceof ModuleComponentSelector)) {
+                        // We only care about substituting for a module, not a project.
+                        return
+                    }
+
+                    // For every org.mozilla.appservices.* module, substitute its version for '+'.
+                    // '+' version tells gradle to resolve the latest available version.
+                    // As long as 'mavenLocal' is in the repositories list, gradle should pick out
+                    // latest published module during dependency resolution phase.
```

Comment:
Hrm, this doesn't seem to work unless I increment the version number in `.build-config.yml`. If the latest published version if `0.52.0` and my local publishing workflow creates a `0.52.0-LOCAL-TIMESTAMP` then gradle will prefer the `0.52.0` version. Maybe I need to tweak the format of the version numbers generated locally.

Reply:
Going to hack around this by forcing the locally-published version number to be astronomically high.